lib/retrieveFileStreams.js

```diff
@@ -0,0 +1,13 @@
+function retrieveFileStreams (conn, fileStreamArray, client_type) {
+  return Promise.all(fileStreamArray.map((file) => {
+    // make sure we are using sftp library
+    if (client_type === "sftp") {
+      // make sure we are returning files and not directories
+      if (file.type === "-") {
+        return conn.get(file.name)
+      }
+    }
```

Comment:
Actually no, because what if one of them is a directory. We only want the fail to happen if the file get doesn't work.